log4j-core/src/main/java/org/apache/logging/log4j/core/async/RingBufferLogEventHandler4.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.core.async;
+
+import com.lmax.disruptor.EventHandler;
+import com.lmax.disruptor.Sequence;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.util.LoaderUtil;
+
+/**
+ * This event handler gets passed messages from the RingBuffer as they become
+ * available. Processing of these messages is done in a separate thread,
+ * controlled by the {@code Executor} passed to the {@code Disruptor}
+ * constructor.
+ */
+class RingBufferLogEventHandler4 implements EventHandler<RingBufferLogEvent> {
```

Comment:
I am not able to understand the `4`-suffixed naming convention here. That is, `RingBufferLogEventHandler4` creates the impression that it is supposed to deliver some Disruptor4-specific functionality. On the contrary, it first attempts to `create()` an instance using Disruptor3, and, on failure, falls back to Disruptor4.

I would have preferred a more accurate and self-explanatory name (e.g., `RBLEHAdapter`) *and* documenting the fact that this class mediates an `EventHandler<RingBufferLogEvent>` between multiple Disruptor major versions, i.e., 3 and 4.

Reply:
I added a comment explain that this class only works with Disruptor 4.x (same with `RingBufferLogEventHandler` whose name and implemented interfaces I can not change for BC).

Reply:
Okay. Thanks for moving Disruptor 3 code out of here. :pray: